site-scrapers/NatickMall.js

```diff
@@ -0,0 +1,102 @@
+const sites = require("../data/sites.json");
+const https = require("https");
+
+module.exports = async function GetAvailableAppointments() {
+    console.log("NatickMall starting.");
+    const webData = await ScrapeWebsiteData();
+    console.log("NatickMall done.");
+    return {
+        ...sites.NatickMall,
+        ...webData,
+    };
+};
+
+async function ScrapeWebsiteData() {
+    const tokenUrl =
+        "https://home.color.com/api/v1/get_onsite_claim?partner=natickmall";
+    // Get a request token
+    const tokenPromise = new Promise((resolve) => {
+        https
+            .get(tokenUrl, (res) => {
+                let body = "";
+                res.on("data", (chunk) => {
+                    body += chunk;
+                });
+                res.on("end", () => {
+                    response = body;
+
+                    resolve(body);
+                });
+            })
+            .on("error", (e) => {
+                console.error(
+                    "Error making token request for the Natick Mall" + e
+                );
+            })
+            .end();
+    });
+    const tokenResponse = await tokenPromise;
+    const responseJson = JSON.parse(tokenResponse);
+    const token = responseJson["token"];
+    const availabilityUrl = `https://home.color.com/api/v1/vaccination_appointments/availability?claim_token=${token}&collection_site=Natick%20Mall`;
+    const availabilityPromise = new Promise((resolve) => {
+        https
+            .get(availabilityUrl, (res) => {
+                let body = "";
+                res.on("data", (chunk) => {
+                    body += chunk;
+                });
+                res.on("end", () => {
+                    response = body;
+
+                    resolve(body);
+                });
+            })
+            .on("error", (e) => {
+                console.error(
+                    "Error making token request for the Natick Mall" + e
+                );
+            })
+            .end();
+    });
+
+    const availabilityResponse = await availabilityPromise;
+    const availability = JSON.parse(availabilityResponse);
+    const results = { availability: {}, hasAvailability: false };
+    // Collect availability count by date
+    availability.reduce((memo, currentValue) => {
+        /* The availability returns and array of appointments like this:
```

Comment:
We just pushed a change today that converts this from an array to an object wrapper with a `results` array inside it:

`{"results":[],"target_date":null}`

Sorry about the breaking change, but we don't expect any further breaking changes in the near future.

Reply:
Thank you!!